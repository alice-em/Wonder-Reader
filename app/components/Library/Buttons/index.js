export { default as CloseLibrary } from './CloseLibrary';
export { default as OpenDirectory } from './OpenDirectory';
export { default as SetParentAsLibrary } from './SetParentAsLibrary';
