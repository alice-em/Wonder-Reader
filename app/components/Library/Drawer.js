import Drawer from '@material-ui/core/Drawer';
import PropTypes from 'prop-types';
import React from 'react';

import Context from '../../context/ContextFactory';

const styles = {
  PaperProps: {
    borderRadius: '0px 0px 0px 15px',
    margin: 'auto',
    // maxWidth: '960px',
  },
};

const CustomDrawer = ({ children, style }) => (
  <Context.Consumer>
    {({ setState, state }) => (
      <div className="Library" style={style}>
        <Drawer
          anchor="top"
          open={state.isLibraryActive}
          onClose={() => setState({ isLibraryActive: false })}
          PaperProps={{ style: styles.PaperProps }}
          variant="temporary"
          transitionDuration={125}
        >
          {children}
        </Drawer>
      </div>
    )}
  </Context.Consumer>
);

CustomDrawer.defaultProps = {
  style: {},
};

CustomDrawer.propTypes = {
  children: PropTypes.node.isRequired,
  style: PropTypes.objectOf(PropTypes.object.isRequired),
};

export default CustomDrawer;
