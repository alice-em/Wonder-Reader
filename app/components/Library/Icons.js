import IconButton from '@material-ui/core/IconButton';
import React from 'react';
import { FaClose, FaFolderOpen, FaLevelUp } from 'react-icons/lib/fa';

const styles = {
  closeButton: {
    background: '#ef5350',
  },
};

const Close = onClick => (
  <IconButton onClick={onClick} color="primary" style={styles.closeButton}>
    <FaClose />
  </IconButton>
);

const FolderOpen = ({ onClick }) => (
  <IconButton onClick={onClick} color="primary">
    <FaFolderOpen />
  </IconButton>
);

const LevelUp = ({ onClick }) => (
  <IconButton onClick={onClick} color="primary">
    <FaLevelUp />
  </IconButton>
);

export { Close, FolderOpen, LevelUp };
