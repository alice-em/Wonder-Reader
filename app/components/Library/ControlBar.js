import React from 'react';

import * as Buttons from './Buttons';

const ControlBar = () => (
  <div>
    <Buttons.OpenDirectory />
    <Buttons.SetParentAsLibrary />
    <Buttons.CloseLibrary />
  </div>
);

export default ControlBar;
