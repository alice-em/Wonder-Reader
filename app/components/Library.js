import Drawer from '@material-ui/core/Drawer';
import PropTypes from 'prop-types';
import React, { Component } from 'react';
import withStyles from '@material-ui/core/styles/withStyles';

import LibraryLayout from './LibraryLayout';

const styles = {
  list: {
    width: 250,
  },
  fullList: {
    width: 'auto',
  },
  PaperProps: {
    borderRadius: '0px 0px 0px 15px',
    margin: 'auto',
    maxWidth: '960px',
  },
};

class Library extends Component {
  state = {
    root: this.props.loadedLibrary,
  };

  updateRoot = (filepath) => {
    this.setState({ root: filepath });
  };

  render() {
    const {
      classes,
      closeDrawer,
      open,
      openComic,
      saveContentDataToMain,
      style,
    } = this.props;
    const { root } = this.state;

    return (
      <div className="Library" style={style}>
        <Drawer
          anchor="top"
          open={open}
          onClose={closeDrawer}
          PaperProps={{ style: styles.PaperProps }}
          variant="temporary"
          transitionDuration={125}
        >
          <div tabIndex={0} role="button" onKeyDown={closeDrawer}>
            <LibraryLayout
              className={classes.list}
              closeLibrary={closeDrawer}
              openComic={openComic}
              root={root}
              saveContentDataToParent={saveContentDataToMain}
              updateRoot={this.updateRoot}
            />
          </div>
        </Drawer>
      </div>
    );
  }
}

Library.defaultProps = {
  loadedLibrary: null,
  style: {},
};

Library.propTypes = {
  classes: PropTypes.object.isRequired, // eslint-disable-line
  closeDrawer: PropTypes.func.isRequired,
  loadedLibrary: PropTypes.string,
  open: PropTypes.bool.isRequired,
  openComic: PropTypes.func.isRequired,
  saveContentDataToMain: PropTypes.func.isRequired,
  style: PropTypes.objectOf(PropTypes.object.isRequired),
};

export { Library };
export default withStyles(styles)(Library);
