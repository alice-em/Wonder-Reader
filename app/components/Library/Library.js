import Drawer from '@material-ui/core/Drawer';
import PropTypes from 'prop-types';
import React from 'react';
import withStyles from '@material-ui/core/styles/withStyles';

import Context from '../../context/ContextFactory';
import Layout from './Layout';

const styles = {
  list: {
    width: 250,
  },
  fullList: {
    width: 'auto',
  },
  PaperProps: {
    borderRadius: '0px 0px 0px 15px',
    margin: 'auto',
    // maxWidth: '960px',
  },
};

const Library = ({
  classes,
  closeDrawer,
  openComic,
  saveContentDataToMain,
  style,
}) => (
  <Context.Consumer>
    {({ state }) => (
      <div className="Library" style={style}>
        <Drawer
          anchor="top"
          open={state.top}
          onClose={closeDrawer}
          PaperProps={{ style: styles.PaperProps }}
          variant="temporary"
          transitionDuration={125}
        >
          <div tabIndex={0} role="button" onKeyDown={closeDrawer}>
            <Layout
              className={classes.list}
              closeLibrary={closeDrawer}
              loadedLibrary={state.loadedLibrary}
              openComic={openComic}
              saveContentDataToParent={saveContentDataToMain}
            />
          </div>
        </Drawer>
      </div>
    )}
  </Context.Consumer>
);

Library.defaultProps = {
  style: {},
};

Library.propTypes = {
  classes: PropTypes.object.isRequired, // eslint-disable-line
  closeDrawer: PropTypes.func.isRequired,
  openComic: PropTypes.func.isRequired,
  saveContentDataToMain: PropTypes.func.isRequired,
  style: PropTypes.objectOf(PropTypes.object.isRequired),
};

export { Library };
export default withStyles(styles)(Library);
