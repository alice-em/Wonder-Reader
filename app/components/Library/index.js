import Drawer from '@material-ui/core/Drawer';
import PropTypes from 'prop-types';
import React, { Component } from 'react';
import withStyles from '@material-ui/core/styles/withStyles';

import Layout from './Layout';

const styles = {
  list: {
    width: 250,
  },
  fullList: {
    width: 'auto',
  },
  PaperProps: {
    borderRadius: '0px 0px 0px 15px',
    margin: 'auto',
    maxWidth: '960px',
  },
};

class Library extends Component {
  state = {
    root: this.props.loadedLibrary,
  };

  render() {
    const { root } = this.state;
    const {
      classes,
      closeDrawer,
      open,
      openComic,
      saveContentDataToMain,
      style,
    } = this.props;

    const updateRoot = (filepath) => {
      this.setState({ root: filepath });
    };

    return (
      <div className="Library" style={style}>
        <Drawer
          anchor="top"
          open={open}
          onClose={closeDrawer}
          PaperProps={{ style: styles.PaperProps }}
          variant="temporary"
          transitionDuration={125}
        >
          <div tabIndex={0} role="button" onKeyDown={closeDrawer}>
            <Layout
              className={classes.list}
              closeLibrary={closeDrawer}
              openComic={openComic}
              root={root}
              saveContentDataToParent={saveContentDataToMain}
              updateRoot={updateRoot}
            />
          </div>
        </Drawer>
      </div>
    );
  }
}

Library.defaultProps = {
  loadedLibrary: null,
  style: {},
};

Library.propTypes = {
  classes: PropTypes.object.isRequired, // eslint-disable-line
  closeDrawer: PropTypes.func.isRequired,
  loadedLibrary: PropTypes.string,
  open: PropTypes.bool.isRequired,
  openComic: PropTypes.func.isRequired,
  saveContentDataToMain: PropTypes.func.isRequired,
  style: PropTypes.objectOf(PropTypes.object.isRequired),
};

export { Library };
export default withStyles(styles)(Library);
