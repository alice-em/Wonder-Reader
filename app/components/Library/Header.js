import AppBar from '@material-ui/core/AppBar';
import React from 'react';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';
import { ThemeProvider as MuiThemeProvider } from '@material-ui/core/styles';

import headerStyle from '../headerStyle';
import ControlBar from './ControlBar';
import { buttonStyle, buttonTheme } from '../Header/Buttons/buttonStyle';

const Header = () => (
  <AppBar position="fixed" >
    <Toolbar>
      <Typography variant="h5" style={headerStyle}>
        Library
      </Typography>
      <MuiThemeProvider theme={buttonTheme}>
        <div style={buttonStyle}>
          <ControlBar />
        </div>
      </MuiThemeProvider>
    </Toolbar>
  </AppBar>
);

export { Header };
export default Header;
