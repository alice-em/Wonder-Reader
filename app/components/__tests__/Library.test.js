import Adapter from 'enzyme-adapter-react-16';
import Enzyme, { shallow } from 'enzyme';
import React from 'react';

import { Library } from '../Library';

Enzyme.configure({ adapter: new Adapter() });

const props = {
  classes: { list: 'classes.list' },
  closeDrawer: jest.fn(),
  loadedLibrary: './',
  open: true,
  openComic: jest.fn(),
  saveContentDataToMain: jest.fn(),
  style: {},
};

describe('Library', () => {
  it('should render', () => {
    const wrapper = shallow(<Library {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
