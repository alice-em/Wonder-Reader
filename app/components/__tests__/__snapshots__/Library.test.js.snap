// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Library should render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Library
    classes={
      Object {
        "list": "classes.list",
      }
    }
    closeDrawer={[MockFunction]}
    loadedLibrary="./"
    open={true}
    openComic={[MockFunction]}
    saveContentDataToMain={[MockFunction]}
    style={Object {}}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <WithStyles(Drawer)
        PaperProps={
          Object {
            "style": Object {
              "borderRadius": "0px 0px 0px 15px",
              "margin": "auto",
              "maxWidth": "960px",
            },
          }
        }
        anchor="top"
        onClose={[MockFunction]}
        open={true}
        transitionDuration={125}
        variant="temporary"
      >
        <div
          onKeyDown={[MockFunction]}
          role="button"
          tabIndex={0}
        >
          <LibraryLayout
            className="classes.list"
            closeLibrary={[MockFunction]}
            openComic={[MockFunction]}
            root="./"
            saveContentDataToParent={[MockFunction]}
            updateRoot={[MockFunction]}
          />
        </div>
      </WithStyles(Drawer)>,
      "className": "Library",
      "style": Object {},
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "PaperProps": Object {
          "style": Object {
            "borderRadius": "0px 0px 0px 15px",
            "margin": "auto",
            "maxWidth": "960px",
          },
        },
        "anchor": "top",
        "children": <div
          onKeyDown={[MockFunction]}
          role="button"
          tabIndex={0}
        >
          <LibraryLayout
            className="classes.list"
            closeLibrary={[MockFunction]}
            openComic={[MockFunction]}
            root="./"
            saveContentDataToParent={[MockFunction]}
            updateRoot={[MockFunction]}
          />
        </div>,
        "onClose": [MockFunction],
        "open": true,
        "transitionDuration": 125,
        "variant": "temporary",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <LibraryLayout
            className="classes.list"
            closeLibrary={[MockFunction]}
            openComic={[MockFunction]}
            root="./"
            saveContentDataToParent={[MockFunction]}
            updateRoot={[MockFunction]}
          />,
          "onKeyDown": [MockFunction],
          "role": "button",
          "tabIndex": 0,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "className": "classes.list",
            "closeLibrary": [MockFunction],
            "openComic": [MockFunction],
            "root": "./",
            "saveContentDataToParent": [MockFunction],
            "updateRoot": [MockFunction],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <WithStyles(Drawer)
          PaperProps={
            Object {
              "style": Object {
                "borderRadius": "0px 0px 0px 15px",
                "margin": "auto",
                "maxWidth": "960px",
              },
            }
          }
          anchor="top"
          onClose={[MockFunction]}
          open={true}
          transitionDuration={125}
          variant="temporary"
        >
          <div
            onKeyDown={[MockFunction]}
            role="button"
            tabIndex={0}
          >
            <LibraryLayout
              className="classes.list"
              closeLibrary={[MockFunction]}
              openComic={[MockFunction]}
              root="./"
              saveContentDataToParent={[MockFunction]}
              updateRoot={[MockFunction]}
            />
          </div>
        </WithStyles(Drawer)>,
        "className": "Library",
        "style": Object {},
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "PaperProps": Object {
            "style": Object {
              "borderRadius": "0px 0px 0px 15px",
              "margin": "auto",
              "maxWidth": "960px",
            },
          },
          "anchor": "top",
          "children": <div
            onKeyDown={[MockFunction]}
            role="button"
            tabIndex={0}
          >
            <LibraryLayout
              className="classes.list"
              closeLibrary={[MockFunction]}
              openComic={[MockFunction]}
              root="./"
              saveContentDataToParent={[MockFunction]}
              updateRoot={[MockFunction]}
            />
          </div>,
          "onClose": [MockFunction],
          "open": true,
          "transitionDuration": 125,
          "variant": "temporary",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <LibraryLayout
              className="classes.list"
              closeLibrary={[MockFunction]}
              openComic={[MockFunction]}
              root="./"
              saveContentDataToParent={[MockFunction]}
              updateRoot={[MockFunction]}
            />,
            "onKeyDown": [MockFunction],
            "role": "button",
            "tabIndex": 0,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "className": "classes.list",
              "closeLibrary": [MockFunction],
              "openComic": [MockFunction],
              "root": "./",
              "saveContentDataToParent": [MockFunction],
              "updateRoot": [MockFunction],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
